src/components/app/content/cardList/card/index.js

```diff
@@ -0,0 +1,53 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import Context from "../../context";
+import './index.css'
+import CardHeader from "./cardHeader";
+import CardBody from "./cardBody";
+
+function Card(props) {
+    const cardData = props.cardData;
+    const cardId = cardData.id;
+
+    const [tempState, setTempState] = useState({header: props.cardData.header, body: props.cardData.body});
+    const [editable, setEditable] = useState(false);
+
+    const cancelEditing = () => setEditable(false);
+
+    const saveCardDataChanges = () => {
+        cancelEditing();
+        props.saveCardData(cardId, tempState);
+    };
+    const restoreCardDataChanges = useCallback(() => {
+        cancelEditing();
+        setTempState({header: cardData.header, body: cardData.body});
+    }, [cardData.header, cardData.body]);
```

Comment:
Why not use a spread operator to copy card object? far more elegant way.
Also, you can check for props.cardData.